app/perfdwarfdiecache.cpp

```diff
@@ -240,6 +240,16 @@ QByteArray demangle(const QByteArray &mangledName)
             if (status == 0)
                 return demangleBuffer = dsymname;
         }
+
+#ifdef HAVE_D_DEMANGLE
+        if (mangledName[0] == '_' && mangledName[1] == 'D') {
+            extern const char* ddemangle(const char*, char*, size_t*);
+            size_t len = demangleBufferLength;
+
+            if (ddemangle(mangledName.constData(), demangleBuffer, &len))
+                return QByteArray(out, len);
```

Comment:
Hi,
I tried this PR and found that `out` must be `demangleBuffer`.
After this change, it works fine for me.

Side note: I personally don't like Docker in a build process (requires root privileges), so I create libddemangle.so myself. Could this be done through the cmake scripts?

Reply:
Oops, it seems I made a change before pushing, it doesn't even compile I think.
The code should be something like:
```
const char* out = ddemangle(mangledName.constData(), demangleBuffer, &len);
if (out)
    return QByteArray(out, len);
```
In most of the cases `out` is `demangleBuffer` but if the symbol is too big the returned string comes from D side.

>Could this be done through the cmake scripts?

Of course, I'll try to do it as soon as I can.